tools/testsys-config/src/lib.rs

```diff
@@ -261,6 +261,14 @@ pub struct GenericVariantConfig {
     pub control_plane_endpoint: Option<String>,
     /// The path to userdata that should be used for Bottlerocket launch
     pub userdata: Option<String>,
+    /// The directory containing Bottlerocket images. For metal, this is the directory containing
```

Comment:
It's pretty awkward the way this config has things that are required for some variants and other things that are required for other variants. I don't suppose there's much we can do about it right now, but it's worth thinking about. There are different configuration "schemas" for different provisioning scenarios.

Reply:
There are 2 things that need to be considered.
* It is nice that we can set a value for `aws` and have it work for `aws-k8s` and `aws-ecs` variants.
* The `reduced_config` fn works really well and simply since all `GenericConfigs` are the same.